### PR TITLE
What's new for 4.7.2

### LINKED ISSUE #4793: What's new in .NET Framework 4.7.2
We need to document the new features available in the .NET Framework 4.7.2 release at https://docs.microsoft.com/en-us/dotnet/framework/whats-new/

A couple of related blog posts:
- https://blogs.msdn.microsoft.com/dotnet/2018/03/08/net-framework-4-7-2-developer-pack-early-access-build-3056-is-available/
- https://blogs.msdn.microsoft.com/dotnet/2018/02/05/announcing-net-framework-4-7-2-early-access-build-3052/

/cc @rpetrusha

### PR DESCRIPTION
**Pull request by rpetrusha**
## What's new for 4.7.2

Fixes #4793 

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/framework/whats-new/index?branch=pr-en-us-4857)

//cc @preetikr

### CODE REVIEW

File: docs/framework/whats-new/index.md [hunk 1]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+- [SQL](#sql472)
+- [WPF](#wpf472)
+- [ClickOnce](#ClickOnce472)
+
+A continuing focus in the .NET Framework 4.7.2 is improved accessibility, which allows an application to provide an appropriate experience for users of Assistive Technology. For information on accessibility improvement in the .NET Framework 4.7.2, see [What's new in accessibility in the .NET Framework](whats-new-in-accessibility.md). 
+
+<a name="core-472" />
+#### Core
+
+The .NET Famework 4.7.2 features a large number of cryptographic enhancements, better decompression support for ZIP archives, and additional collection APIs.
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: Famework -> Framework

File: docs/framework/whats-new/index.md [hunk 2]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+**New overloads of RSA.Create and DSA.Create**
+
+The <xref:System.Security.Cryptography.DSA.Create(System.Security.Cryptography.DSAParamteters)?displayProperty=nameWithType> and <xref:System.Security.Cryptography.RSA.Create(System.Security.Cryptography.RSAParamteters)?displayProperty=nameWithType> methods let you supply key parameters when instantiated a new <xref:System.Security.Cryptography.DSA> or <xref:System.Security.Cryptography.RSA> key. They allow you to replace code like the following:
+
+```csharp
+// Before .NET Framework 4.7.2
+using (RSA rsa = RSA.Create())
+{
+   rsa.ImportParameters(rsaParameters);
+   // Other code to execute using the rsa instance.
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: uppercase the algorithm name?

File: docs/framework/whats-new/index.md [hunk 3]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+}
+``` 
+```vb
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: nit: it's good to separate code blocks from text by using new lines. It renders fine but it's easier to read here.

File: docs/framework/whats-new/index.md [hunk 4]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+   Using pbkdf2 = New Rfc2898DeriveBytes(password, SaltSize, iterations, algorithm)
+      salt = pbkdf2.Salt
+      Return pbkdf2.GetBytes(DerivedValueSize)
+   End Using
+End Function 
+```
+
+**Support for ephemeral keys**
+
+PFX import can optionally load private keys directly from memory, bypassing the hard drive. When the new <xref:System.Security.Cryptography.X509Certificates.X509KeyStorageFlags.EphemeralKeySet?displayProperty=nameWithType> flag is specified in an <xref:System.Security.Cryptography.X509Certificates.X509Certificate2> constructor or one of the overloads of the <xref:System.Security.Cryptography.X509Certificates.X509Certificate2.Import%2A?displayProperty=nameWithType> method, the private keys will be loaded as ephemeral keys. This prevents the keys from being visible on the disk. However:
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: define PFX?

File: docs/framework/whats-new/index.md [hunk 5]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+}
+``` 
+```vb
+' Before .NET Framework 4.7.2
+Using rsa = RSA.Create(rsaParameters)
+   ' Other code to execute using the rsa instance.
+End Using
+``` 
+
+The <xref:System.Security.Cryptography.DSA.Create(System.Int32)?displayProperty=nameWithType> and <xref:System.Security.Cryptography.RSA.Create(System.Int32)?displayProperty=nameWithType> methods let you generate new <xref:System.Security.Cryptography.DSA> or <xref:System.Security.Cryptography.RSA> keys with a specific keysize. For example:
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: keysize -> key size

File: docs/framework/whats-new/index.md [hunk 6]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+
+Starting with the .NET Framework 4.7.2, a diagnostic assistant can locate the owners of a given <xref:Windows.UI.Xaml.ResourceDictionary>. (The feature is for use by diagnostic assistants and not by production applications.) Whenever a change is made to a <xref:Windows.UI.Xaml.ResourceDictionary>, WPF automatically finds all [DynamicResource](../wpf/advanced/dynamicresource-markup-extension.md) references that might be affected by the change. 
+
+A diagnostic assistant such as Visual Studio’s “Edit-and-Continue” facility may want extend this to handle [StaticResource](../wpf/advanced/staticresource-markup-extension.md) references. The first step in this process is to find the owners of the dictionary; that is, to find all the objects whose `Resources` property refers to the dictionary (either directly, or indirectly via the <xref:System.Windows.ResourceDictionary.MergedDictionaries?displayProperty=nameWithType> property). Three new static methods implemented on the <xref:System.Windows.Diagnostics.ResourceDictionaryDiagnostics?displayProperty=nameWithType> class, one for each of the base types that has a `Resources` property,support this step:
```
Comment (mairaw) [inline docs/framework/whats-new/index.md]: nit: add space after last comma

**Review by mairaw — CHANGES_REQUESTED** [review]
This looks great @rpetrusha. Left a few comments for you to take care of. Links are still broken but I have PR in place for 4.7.2.

File: docs/framework/whats-new/index.md [hunk 7]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+public class StaticResourceResolvedEventArgs : EventArgs
+{
+   public Object TargetObject { get; private set; }
+
+   public Object TargetProperty { get; private set; }
+
+   public ResourceDictionary ResourceDictionary { get; private set; }
+
+   public object ResourceKey { get; private set; }
+}
```
Comment (svick) [inline docs/framework/whats-new/index.md]: If this is meant to show just the public API, then I think all the `private set;` are not necessary. You can just make the properties appear as if they were `get`-only auto-properties.

File: docs/framework/whats-new/index.md [hunk 8]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+```
+
+The event is not raised (and its `add` method is ignored) unless <xref:System.Windows.Diagnostics.VisualDiagnostics> is enabled and the [`ENABLE_XAML_DIAGNOSTICS_SOURCE_INFO`](xref:System.Windows.Diagnostics.VisualDiagnostics.GetXamlSourceInfo%2A) environment variable is set.
```
Comment (svick) [inline docs/framework/whats-new/index.md]: I think it's more correct to call `add` an accessor, not a method.

File: docs/framework/whats-new/index.md [hunk 9]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+
+<a name="clickonce472" />
+#### ClickOnce
+
+HDPI-aware applications for Windows Forms, Windows Presentation Foundation (WPF), and Visual Studio Tools for Office (VSTO) can all be deployed by using ClickOnce. If the following entry is found in teh application manifest, deployment will succeed under .NET Framework 4.7.2:
+
+```xml
+<windowsSettings>
+      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
+    </windowsSettings>
```
Comment (svick) [inline docs/framework/whats-new/index.md]: Nit: inconsistent indentation

File: docs/framework/whats-new/index.md [hunk 10]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+   public object ResourceKey { get; private set; }
+}
+```
+
+The event is not raised (and its `add` method is ignored) unless <xref:System.Windows.Diagnostics.VisualDiagnostics> is enabled and the [`ENABLE_XAML_DIAGNOSTICS_SOURCE_INFO`](xref:System.Windows.Diagnostics.VisualDiagnostics.GetXamlSourceInfo%2A) environment variable is set.
+
+<a name="clickonce472" />
+#### ClickOnce
+
+HDPI-aware applications for Windows Forms, Windows Presentation Foundation (WPF), and Visual Studio Tools for Office (VSTO) can all be deployed by using ClickOnce. If the following entry is found in teh application manifest, deployment will succeed under .NET Framework 4.7.2:
```
Comment (svick) [inline docs/framework/whats-new/index.md]: Typo: teh

**Comment by rpetrusha** [comment]
Thanks for the thorough review and the comments, @svick! I'll address them in the next round of revisions.

File: docs/framework/whats-new/index.md [hunk 11]
```diff
@@ -36,23 +37,381 @@ This article does not provide comprehensive information about each new feature a
...
+```vb
+' Before .NET Framework 4.7.2
+Using rsa = RSA.Create()
+   rsa.ImportParameters(rsaParameters)
+   ' Other code to execute using the rsa instance.
+End Using
+``` 
+with code like this:
+```csharp
+// Before .NET Framework 4.7.2
```
[the remark this answers is not in the document]
Comment (rpetrusha) [reply docs/framework/whats-new/index.md]: Thanks for catching this, @svick.

File: docs/framework/whats-new/index.md (hunk 4 above)
Comment (rpetrusha) [reply docs/framework/whats-new/index.md]: I think I'll leave it undefined, @mairaw.

**Comment by rpetrusha** [comment]
Thanks for the reviews, @mairaw and @svick. I've addressed your comments.

**Review by mairaw — APPROVED** [review]

**Pull request merged**